boards: shields: esp_8266: Add std headers overlay (#26817)
zephyrproject-rtos/zephyr: boards/shields/esp_8266/doc/index.rst

This shield requires a board which provides a configuration that allows an
UART interface. (see :ref:`shields` for more details).

.. note::

>    Sometimes boards declare standard heades like Arduino R3 but not define all

Reviewer: ```suggestion
   Sometimes boards declare standard headers like Arduino R3 but not define all
```

Author: Done.

        - ESP-8266 bootloader won't send garbage. Try connect at 74880 bps if
          you module have 26MHz crystal to detect boot fails.

.. note::
	Boards that already have a network interface:  Check network

> 	documentation to undestand how proper configure both interfaces.  To

Reviewer: ```suggestion
	documentation to understand how properly configure both interfaces.  To
```

Author: Done.

        - ESP-8266 bootloader won't send garbage. Try connect at 74880 bps if
          you module have 26MHz crystal to detect boot fails.

.. note::
	The shield configuration was designed to be used with multiple network
	interfaces.  This means, boards with any other network interface can
	be simultaneous enabled.  User must be check network documentation to
	avoid problem when running samples.  For instance, boards with ethernet
	automatically define that interface as default.  Make sure to select
	WiFi interface as the default interface when build examples.
	As general rule, keep only WiFi interface enabled at
	Networking -> Link Layer Options.  This will ensure samples works as

> 	expected.

Reviewer: I don't quite understand this note:
You first mention that shield is designed to work with multiple network interfaces and you end up stating that they should be disabled.

Author: reworked.

Supported variations
====================

The below table suggests shield variation accordingly with end user

> application.  The default shield esp_8266 expects a board overlay.  The

Reviewer: ```suggestion
application. When a standard connector (arduino, mikrobus) is available on board, user should 
select the matching shield configuration. When esp_8266 shield is used with a board that doesn't
feature a standard connector, a dedicated <board>.overlay file should be provided.
```

Author: Done.

remaining configurations should be used based on the board standard headers

> available.  Any variation selects automatically the especific overlay that
> define right configuration.

Reviewer: I think this last sentence should be removed, since what you describe is valid for all the shield.
```suggestion
available.
```

Author: done.
